### ISSUE TITLE
"spacemacs/helm-project-smart-do-search" is VERY slow in large projects.

### ISSUE THREAD

**Issue by synic** [syl20bnr/spacemacs#10600]
#### Description :octocat:
`spacemacs/helm-project-smart-do-search` is VERY slow in large projects.

#### Reproduction guide :beetle:
- Start Emacs
- Open a large project
- Type `SPC /`
- Type in a search term

*Observed behaviour:* :eyes: :broken_heart:
If I type `class SomethingViewSet`, I get a CPU spike, and the search result can take up to 10 seconds to appear.

*Expected behaviour:* :heart: :smile:
It should be a LOT faster.  If I go into the terminal and type `ag "class SomethingViewSet"`, it takes less than
a second to appear.

#### System Info :computer:
- OS: darwin
- Emacs: 25.2.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. d4017b1ca)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(go swift csv lua nginx sql auto-completion emacs-lisp git yaml javascript html markdown major-modes python org osx python ipython-notebook django syntax-checking spell-checking colors spacemacs-layouts vimscript helm
    (version-control :variables version-control-diff-tool 'diff-hl)
    (c-c++ :variables c-c++-enable-clang-support t)
    (shell :variables shell-default-height 30 shell-default-position 'bottom))
```
- System configuration features: NOTIFY ACL LIBXML2 ZLIB TOOLKIT_SCROLL_BARS

**Comment by synic** [comment]
Further investigation leads me to the fact that this is causing the following to run:

`/usr/local/bin/ag --nocolor --nogroup "(?=.*class.*)(?=.*SomethingViewSet.*)"`, which is indeed slow.

How can I tell it to search for an exact string instead of a pattern?

**Comment by thanhvg** [comment]
pass -Q with your search query you will have literal search, https://github.com/syohex/emacs-helm-ag

**Comment by synic** [comment]
@thanhvg This does not have the intended consequences, it causes to search for `"(?=.*class.*)(?=.*SomethingViewSet.*)"` as a literal, which will never be found.

**Comment by synic** [comment]
Ah, you cannot add it to `helm-ag-command-option`, you have to add it to `helm-ag-base-command`, which is a little weird.

Thanks!

**Comment by thanhvg** [comment]
This is how i search for a literal  string: abc
type: `SPC /` then type `-Q abc`

I have no change in dot file

**Comment by rafa-unx** [comment]
I had the same behavior when searching with `SPC /`. As inexperienced spacemacs user, it was not easy for me to find a solution so I would like to share in case it helps somebody.

Searching with `SPC /` was always slow when typing (it used to take about 1 minute every time I pressed a key), but it was not slow when pasting/selecting text. To solve it, I installed ripgrep and added the following in `(defun dotspacemacs/user-config ()`

```
(evil-leader/set-key "/" 'spacemacs/helm-project-do-ag)
(setq helm-ag-base-command "rg --vimgrep --no-heading --smart-case")
```
Now searching is working as fast as I expect it to be.
